### PR TITLE
Core/82 provide better match options

### LINKED ISSUE #82: Modify commands in `exports.match` to avoid false matches for similar commands
This is quite subjective and hence I haven't made a pull request for this. The issue is within the `exports.match` method's usage.

Most modules check for a command match with this `text.startsWith(commandPrefix + 'anim');`. The code is only checking for `/anim` as a prefix. The user can potentially pass in `/animatedHacks` and still have the module run; or pretty much anything that starts with `/anim`

Main side-effect is, this has limited the possible number of module commands within an instance of Kassy. For example, I can't create a module with the command `/anime` because all queries will be forwarded to `/anim` instead. 

A simple fix is to append a whitespace after the command, like this `text.startsWith(commandPrefix + 'anim ');`. I tried to add this in all modules, but some modules handle no queries with a specific message as well.

For example, `/quote` responds with a `Um.. what do you expect me to do?` message.

So is this fix the right thing to do here? Or is there a better solution?

This is regarding the main Kassy source, I know I can do whatever in my own fork :)

### PR DESCRIPTION
**Pull request by mrkno**
resolves #82 and implements work towards #71

### CODE REVIEW

File: core/core_modules/help.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ shortSummary = function(prefix) {
 	return constructHelpMessage(help, this.loadedModules, context, prefix);
 },
 
-longDescription = function(moduleName, prefix) {
```
Comment (Awarua-) [inline core/core_modules/help.js]: Commit noise

File: core/core_modules/creator.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-exports.match = function(text, commandPrefix) {
-	return text === commandPrefix + 'creator';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/creator.js]: Is Match still required in the new api

File: core/core_modules/disable.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 var disabled = false;
 
-exports.match = function(text, commandPrefix) {
-	return disabled || text === commandPrefix + 'disable';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/disable.js]: Is Match still required in the new api

File: core/core_modules/help.js [hunk 4]
```diff
@@ -59,29 +59,27 @@ longDescription = function(moduleName, prefix) {
 	return help;
 };
 
-exports.match = function(text, commandPrefix) {
-	return text === commandPrefix + this.packageInfo.name
-		|| text === commandPrefix + 'help'
-		|| (text.startsWith(commandPrefix + this.packageInfo.name + ' '))
-		|| text.startsWith(commandPrefix + 'help ');
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/help.js]: Is Match still required in the new api

File: core/core_modules/issue.js [hunk 5]
```diff
@@ -111,22 +110,23 @@ ifStringNotEmpty = function(str) {
...
-        ['debugLevel can be basic, detail or full', 'defaults to basic if not specified']];
+    return [
+        [commandPrefix + 'issue "<title>" <debugLevel>', 'Posts an issue to Github.', 'Posts an issue to Github. DebugLevel can be basic, detail or full, defaults to basic if notspecified.'],
+        [commandPrefix + 'issue "<title>" "<description>" <debugLevel>', 'Posts an issue to Github with a description.', 'Posts an issue to Github with a description. DebugLevel can be basic, detail or full, defaults to basic if not specified.']
+    ];
 };
 
-exports.match = function(text, commandPrefix) {
-    return text.startsWith(commandPrefix + command);
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/issue.js]: Is Match still required in the new api

File: core/core_modules/kpm.js [hunk 6]
```diff
@@ -322,12 +355,12 @@ var git = require.once('../git.js'),
         });
     };
 
-exports.match = function (text, commandPrefix) {
-    return text.startsWith(commandPrefix + 'kpm');
+exports.match = function (event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/kpm.js]: Is Match still required in the new api

File: core/core_modules/ping.js [hunk 7]
```diff
@@ -11,8 +11,8 @@
 
 var os = require("os");
 
-exports.match = function(text, commandPrefix) {
-    return text === commandPrefix + 'ping';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/ping.js]: Is Match still required in the new api

File: core/core_modules/restart.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
-exports.match = function(text, commandPrefix) {
-	return text === commandPrefix + 'restart';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/restart.js]: Is Match still required in the new api

File: core/core_modules/shutdown.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 var shutdownResponses = ['Good Night', 'I don\'t blame you.', 'There you are.', 'Please.... No, Noooo!'];
 
-exports.match = function (text, commandPrefix) {
-  return text === commandPrefix + 'shutdown';
+exports.match = function (event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/shutdown.js]: Is Match still required in the new api

File: core/core_modules/update.js [hunk 10]
```diff
@@ -59,14 +59,15 @@ var git = require.once('../git.js'),
 		}, updatePeriod);
 	};
 
-exports.match = function(text, commandPrefix) {
-	return text === commandPrefix + 'update';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/update.js]: Is Match still required in the new api

File: core/core_modules/uptime.js [hunk 11]
```diff
@@ -16,8 +16,8 @@ difference = function (dateFrom, dateTo) {
     return diff;
 };
 
-exports.match = function(text, commandPrefix) {
-    return text === commandPrefix + 'uptime';
+exports.match = function(event, commandPrefix) {
```
Comment (Awarua-) [inline core/core_modules/uptime.js]: Is Match still required in the new api

File: core/modules/modules.js [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+ * 		Matthew Knox
+ *
+ * License:
+ *		MIT License. All code unless otherwise specified is
+ *		Copyright (c) Matthew Knox and Contributors 2015.
+ */
+
+var fs              = require('fs'),
+    path            = require('path'),
+	loaders			= [require.once('./kassyModule.js')],
```
Comment (Awarua-) [inline core/modules/modules.js]: indentation

File: core/modules/modules.js [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    files           = require.once('./../files.js'),
+    config          = require('./../config.js'),
+    modulesDir      = 'modules',
+    descriptor      = 'kassy.json',
+	conflict		= 1;
```
Comment (Awarua-) [inline core/modules/modules.js]: indentation

File: core/modules/modules.js [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+exports.listModules = function (disabled) {
+	var modules = {};
```
Comment (Awarua-) [inline core/modules/modules.js]: indentation

File: core/platform.js [hunk 15]
```diff
@@ -132,19 +135,21 @@ Platform.prototype.start = function() {
 
     // Load core modules
     console.warn('Loading core components...');
-    var m = this.modules.listCoreModules();
+	var coreLoader = require.once('./modules/core.js');
```
Comment (Awarua-) [inline core/platform.js]: indentation

File: core/core_modules/help.js (hunk 1 above)
Comment (mrkno) [inline core/core_modules/help.js]: eh

File: core/core_modules/creator.js (hunk 2 above)
Comment (mrkno) [inline core/core_modules/creator.js]: yes

File: core/core_modules/disable.js (hunk 3 above)
Comment (mrkno) [inline core/core_modules/disable.js]: yes

File: core/core_modules/help.js (hunk 4 above)
Comment (mrkno) [inline core/core_modules/help.js]: yes

File: core/core_modules/issue.js (hunk 5 above)
Comment (mrkno) [inline core/core_modules/issue.js]: yes

File: core/core_modules/kpm.js (hunk 6 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: yes

File: core/core_modules/ping.js (hunk 7 above)
Comment (mrkno) [inline core/core_modules/ping.js]: yes

File: core/core_modules/restart.js (hunk 8 above)
Comment (mrkno) [inline core/core_modules/restart.js]: yes

File: core/core_modules/shutdown.js (hunk 9 above)
Comment (mrkno) [inline core/core_modules/shutdown.js]: yes

File: core/core_modules/update.js (hunk 10 above)
Comment (mrkno) [inline core/core_modules/update.js]: yes

File: core/core_modules/uptime.js (hunk 11 above)
Comment (mrkno) [inline core/core_modules/uptime.js]: yes

File: core/modules/modules.js [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+            modules[t] = m[key];
+            modules[t].__loaderUID = i;
+        }
+    }
+    return modules;
+};
+
+exports.loadModule = function (module) {
+    try {
+        console.write("Loading module '" + module.name + "'... " + (console.isDebug() ? "\n" : ""));
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l39)

File: core/modules/modules.js [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+ * Written By:
+ * 		Matthew Knox
+ *
+ * License:
+ *		MIT License. All code unless otherwise specified is
+ *		Copyright (c) Matthew Knox and Contributors 2015.
+ */
+
+var fs              = require('fs'),
+    path            = require('path'),
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: ["path" is defined but never used](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l13)

File: core/modules/modules.js [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    loaders         = [require.once('./kassyModule.js')],
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Multiple spaces found before '='.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l14)

File: core/modules/modules.js [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+    files           = require.once('./../files.js'),
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Multiple spaces found before '='.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l15)

File: core/modules/modules.js [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+exports.loadModule = function (module) {
+    try {
+        console.write("Loading module '" + module.name + "'... " + (console.isDebug() ? "\n" : ""));
+        var m = loaders[module.__loaderUID].loadModule(module);
+        console.info(console.isDebug() ? "Loading Succeeded" : "\t[DONE]");
+        return m;
+    }
+    catch (e) {
+        console.error(console.isDebug() ? "Loading Failed" : "\t[FAIL]");
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l45)

File: core/modules/modules.js [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+ *		Copyright (c) Matthew Knox and Contributors 2015.
+ */
+
+var fs              = require('fs'),
+    path            = require('path'),
+    loaders         = [require.once('./kassyModule.js')],
+    files           = require.once('./../files.js'),
+    config          = require('./../config.js'),
+    modulesDir      = 'modules',
+    descriptor      = 'kassy.json',
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: ["descriptor" is defined but never used](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l18)

File: core/modules/modules.js (hunk 17 above)
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Multiple spaces found before '='.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l13)

File: core/modules/modules.js [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+    conflict        = 1;
+
+exports.listModules = function (disabled) {
+    var modules = {};
+    for (var i = 0; i < loaders.length; i++) {
+        var m = loaders[i].listModules(disabled);
+        for (var key in m) {
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l25)

File: core/modules/modules.js [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+                t = key + conflict++;
+            }
+            modules[t] = m[key];
+            modules[t].__loaderUID = i;
+        }
+    }
+    return modules;
+};
+
+exports.loadModule = function (module) {
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Function 'anonymous' has a complexity of 5.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l37)

File: core/modules/modules.js (hunk 21 above)
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Multiple spaces found before '='.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529918277/issues/source?bid=3282987&fileBranchId=3282998#l18)

File: core/modules/modules.js [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
```
Comment (mrkno) [inline core/modules/modules.js]: Codacy Issue found: [Function 'anonymous' has a complexity of 5.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926650/issues/source?bid=3282987&fileBranchId=3282998#l32)

File: modules/associate/associate.js [hunk 25]
```diff
@@ -43,30 +44,32 @@ exports.printAssociations = function(api, event) {
...
 	api.sendMessage('Associations cleared.', event.thread_id);
 };
 
-exports.run = function(api, event) {
-	if (!event.body.startsWith(api.commandPrefix + 'associate')) {
-		return exports.match(event.body, api.commandPrefix, event.thread_id, null, api);
+exports.run = function (api, event) {
+    var isCommand = event.arguments[0] === api.commandPrefix + 'associate';
+	if (isCommand && event.arguments.length === 1) {
+		return printAssociations(api, event);
```
Comment (mrkno) [inline modules/associate/associate.js]: Codacy Issue found: ["printAssociations" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l55)

File: modules/associate/associate.js [hunk 26]
```diff
@@ -43,30 +44,32 @@ exports.printAssociations = function(api, event) {
...
-		return exports.clear(api, event);
-	}
-
-	var spl = event.body.split('"');
-	if (spl.length !== 3 && spl.length !== 5)  {
-		api.sendMessage('WTF are you doing????!', event.thread_id);
-		return;
+    
+	if (isCommand && event.arguments[1] === 'clear') {
+		return clear(api, event);
```
Comment (mrkno) [inline modules/associate/associate.js]: Codacy Issue found: ["clear" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l59)

File: modules/associate/associate.js [hunk 27]
```diff
@@ -43,30 +44,32 @@ exports.printAssociations = function(api, event) {
 	api.sendMessage(message, event.thread_id);
 };
 
-exports.clear = function(api, event) {
-	this.config[event.thread_id] = {};
+clear = function(api, event) {
```
Comment (mrkno) [inline modules/associate/associate.js]: Codacy Issue found: ["clear" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l47)

File: modules/associate/associate.js [hunk 28]
```diff
@@ -1,40 +1,41 @@
...
+	return !!event.__associateCmd;
 };
 
-exports.toggleAssociation = function(thread, hook, text) {
-	hook = hook.toLowerCase();
-	if (this.config[thread] && this.config[thread][hook] && !text) {
-		delete this.config[thread][hook];
-		return false;
-	}
+toggleAssociation = function(thread, hook, text) {
```
Comment (mrkno) [inline modules/associate/associate.js]: Codacy Issue found: ["toggleAssociation" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l21)

File: modules/associate/associate.js [hunk 29]
```diff
@@ -43,30 +44,32 @@ exports.printAssociations = function(api, event) {
 	api.sendMessage(message, event.thread_id);
 };
 
-exports.clear = function(api, event) {
-	this.config[event.thread_id] = {};
+clear = function(api, event) {
+	exports.config[event.thread_id] = {};
```
Comment (mrkno) [inline modules/associate/associate.js]: Codacy Issue found: [Identifier 'thread_id' is not in camel case.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926667/issues/source?bid=3282987&fileBranchId=3282998#l48)

File: core/shim.js [hunk 30]
```diff
@@ -85,3 +85,14 @@ exports.createEvent = function(thread, senderId, senderName, message) {
...
+exports.createEventHelpers = function(event) {
+    event.arguments = event.body.match(/(?:[^\s"]+|"[^"]*")+/g);
+    if (event.arguments === null) {
+        event.arguments = [''];
+    }
+    event.arguments_body = event.body.substr(event.arguments[0].length + 1);
+    for (var i = 0; i < event.arguments.length; i++) {
+        event.arguments[i] = event.arguments[i].replace(/(^["])|(["]$)/g, '');
+    }
+};
```
Comment (mrkno) [inline core/shim.js]: Codacy Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926658/issues/source?bid=3282987&fileBranchId=3282998#l98)

File: core/shim.js [hunk 31]
```diff
@@ -85,3 +85,14 @@ exports.createEvent = function(thread, senderId, senderName, message) {
 		body: message
 	};
 };
+
+exports.createEventHelpers = function(event) {
+    event.arguments = event.body.match(/(?:[^\s"]+|"[^"]*")+/g);
+    if (event.arguments === null) {
+        event.arguments = [''];
+    }
+    event.arguments_body = event.body.substr(event.arguments[0].length + 1);
```
Comment (mrkno) [inline core/shim.js]: Codacy Issue found: [Identifier 'arguments_body' is not in camel case.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926658/issues/source?bid=3282987&fileBranchId=3282998#l94)

Comment (mrkno) [inline core/shim.js]: Codacy Issue found: [Identifier 'arguments_body' is not in camel case.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926658/issues/source?bid=3282987&fileBranchId=3282998#l94)

File: core/core_modules/kpm.js [hunk 32]
```diff
@@ -220,67 +216,104 @@ var git = require.once('../git.js'),
...
+    },
 
-                    var instDir = path.resolve('./modules/kpm_' + kj.safeName);
-                    fs.copy(dir, instDir, function (err) {
-                        if (err) {
-                            console.debug(err);
-                            api.sendMessage('An unknown error occurred while installing "' + kj.name + '".', event.thread_id);
-                            cleanup();
-                            return;
-                        }
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529926637/issues/source?bid=3282987&fileBranchId=3282998#l292)

File: modules/tictactoe/tictactoe.js [hunk 33]
```diff
@@ -148,7 +148,7 @@ function checkWin(game, player) {
         if (row[0] === piece && row[1] === piece && row[2] === piece) {
             return player;
         }
-    }   
+    }
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Expected space or tab after "//" in comment.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962506/issues/source?bid=3282987&fileBranchId=3282998#l153)

File: modules/tictactoe/tictactoe.js [hunk 34]
```diff
@@ -168,24 +168,18 @@ function checkWin(game, player) {
...
 };
 
 exports.run = function(api, event) {
-    "use strict"
+    "use strict";
     var command = event.body,
         commandPrefix = api.commandPrefix;
 
-    //If we are creating a new game
+    // If we are creating a new game
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Strings must use singlequote.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962506/issues/source?bid=3282987&fileBranchId=3282998#l183)

File: core/core_modules/kpm.js [hunk 35]
```diff
@@ -43,14 +43,18 @@ var git = require.once('../git.js'),
...
 
                 for (var i = 0; i < args.length; i++) {
                     var url = args[i];
-                    if (url.startsWith('http') || url.startsWith('ssh')) {
-                        install.call(this, url, api, event);
+                    if (url.startsWith('ssh') || url.endsWith('.git')) {
+                        gitInstall.call(this, url, api, event);
+                        continue;
+                    }
+                    else if (url.startsWith('http') && (url.endsWith('.coffee') || url.endsWith('.js'))) {
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l57)

File: core/core_modules/kpm.js [hunk 36]
```diff
@@ -220,67 +218,104 @@ var git = require.once('../git.js'),
...
+                cleanup();
+            });
+        }
+        catch (e) {
+            console.critical(e);
+            api.sendMessage('Could not install "' + name + '".', event.thread_id);
+            cleanup();
+        }
+    },
+
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["gitInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l271)

File: core/core_modules/kpm.js [hunk 37]
```diff
@@ -166,48 +170,42 @@ var git = require.once('../git.js'),
...
         // unload the current version
-        cfg.saveModuleConfig(module.name);
         this.loadedModules = this.loadedModules.filter(function (value) {
-            if (value.name === module.name) {
-                if (value.unload) {
-                    value.unload();
-                }
-                return false;
-            }
-            return true;
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l206)

File: core/core_modules/kpm.js [hunk 38]
```diff
@@ -220,67 +218,104 @@ var git = require.once('../git.js'),
...
+    },
 
-                    var instDir = path.resolve('./modules/kpm_' + kj.safeName);
-                    fs.copy(dir, instDir, function (err) {
-                        if (err) {
-                            console.debug(err);
-                            api.sendMessage('An unknown error occurred while installing "' + kj.name + '".', event.thread_id);
-                            cleanup();
-                            return;
-                        }
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l294)

File: core/core_modules/kpm.js (hunk 37 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962458/issues/source?bid=3282987&fileBranchId=3282998#l206)

File: core/modules/kassyModule.js [hunk 39]
```diff
@@ -0,0 +1,149 @@
...
+    };
+},
+
+getFunctionParameterNames = function (func) {
+    // http://stackoverflow.com/questions/9091838/get-function-parameter-names-for-interface-purposes
+    var f = func.toString();
+    return f.match(/\(.*?\)/)[0].replace(/[()]/gi, '').replace(/\s/gi, '').split(',');
+};
+
+exports.loadModule = function (module) {
```
Comment (mrkno) [inline core/modules/kassyModule.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962470/issues/source?bid=3282987&fileBranchId=3282998#l115)

File: core/shim.js (hunk 31 above)
Comment (mrkno) [inline core/shim.js]: Codacy Issue found: [Identifier 'arguments_body' is not in camel case.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529962479/issues/source?bid=3282987&fileBranchId=3282998#l94)

File: core/core_modules/kpm.js [hunk 40]
```diff
@@ -220,67 +220,104 @@ var git = require.once('../git.js'),
...
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976242/issues/source?bid=3282987&fileBranchId=3282998#l296)

File: core/core_modules/kpm.js [hunk 41]
```diff
@@ -220,67 +220,104 @@ var git = require.once('../git.js'),
...
+                cleanup();
+            });
+        }
+        catch (e) {
+            console.critical(e);
+            api.sendMessage('Could not install "' + name + '".', event.thread_id);
+            cleanup();
+        }
+    },
+
```
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["gitInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976242/issues/source?bid=3282987&fileBranchId=3282998#l273)

File: core/core_modules/kpm.js (hunk 35 above)
Comment (mrkno) [inline core/core_modules/kpm.js]: Codacy Issue found: ["scriptInstall" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976242/issues/source?bid=3282987&fileBranchId=3282998#l57)

File: modules/tictactoe/tictactoe.js [hunk 42]
```diff
@@ -215,22 +217,21 @@ exports.run = function(api, event) {
...
             game = findGame(event.thread_id, event.sender_name.trim());
 
         if (!game) {
-            api.sendMessage("You're not even playing " + event.sender_name.trim(), event.thread_id);
+            api.sendMessage('You\'re not even playing ' + event.sender_name.trim(), event.thread_id);
             return;
         }
 
         if (move.length !== 2) {
-            api.sendMessage("Um. Is that even a coordinate?");
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976290/issues/source?bid=3282987&fileBranchId=3282998#l232)

File: modules/tictactoe/tictactoe.js [hunk 43]
```diff
@@ -50,35 +58,35 @@ function findGame(thread, player) {
...
         board = game.board;
 
     if (player === game.noughts) {
         piece = "O";
     }
 
-    //It's not even your turn
+    // It's not even your turn
     if (player !== game.turn) {
-        return "Invalid Move! (it's not your turn " + player + ")";
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976290/issues/source?bid=3282987&fileBranchId=3282998#l78)

File: modules/tictactoe/tictactoe.js [hunk 44]
```diff
@@ -255,24 +256,24 @@ exports.run = function(api, event) {
...
     }
 
-    //If we are surrendering
-    if (command.startsWith(commandPrefix + "surrender")) {
+    // If we are surrendering
+    if (command.startsWith(commandPrefix + 'surrender')) {
         var game = findGame(event.thread_id, event.sender_name.trim());
 
         if (!game) {
-            api.sendMessage("You're not even in a game! Why are you surrendering?");
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976290/issues/source?bid=3282987&fileBranchId=3282998#l271)

File: modules/tictactoe/tictactoe.js [hunk 45]
```diff
@@ -255,24 +256,24 @@ exports.run = function(api, event) {
...
+            api.sendMessage('You\'re not even in a game! Why are you surrendering?');
             return;
         }
 
         removeGame(game);
-        
-        api.sendMessage(event.sender_name.trim() + " surrenders! " + (event.sender_name.trim() !== game.noughts ? game.noughts : game.crosses) + " wins!");
+
```
Comment (mrkno) [inline modules/tictactoe/tictactoe.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2529976290/issues/source?bid=3282987&fileBranchId=3282998#l277)

**Comment by Awarua-** [comment]
Approved

**Pull request merged**